### ISSUE TITLE
Handle amended filings from netfile.

### ISSUE THREAD

**Issue by mikeubell** [caciviclab/disclosure-backend#49]
Campaigns can file amended documents.  These appear in Socrata as new filings with `Report_num > 0`.  In general only the items with the MAX(Report_num) for any particular filing period should be included in any data displayed.  This makes it hard to pull things straight out of Socrata.

**Comment by rkiddy** [comment]
This is a big problem for analysis tools, as you will probably be seeing, and there will be effects of this all through the system.

I ended up doing a few things.
1) create another column on the FILER_FILINGS table to hold the maximum amend_id value.
2) create another set of tables to store only the latest amended versions.

This second step may sound like overkill, but I was already creating copies. My process was to first import the data from the TSV files into tables with all varchar and text columns. These would be the version 0 tables. Then I would check and correct the data and import the data into properly types columns, thereby creating version 1 tables. Then I would pull latest amend filings out to a version2 set of the tables.

Note that data shows up in earlier amendments and then sometimes may seem to disappear in later amendments. Basically, at its simplest, filers might move money between different filings. This would be interesting to track, but it would not be easy.

**Comment by mikeubell** [comment]
Having tables that only contain the latest amendments is the thing to do.  Whether there is a version of the database that contains all the filings really depends on what you are doing. I think for public visualization only the most recent valid data is interesting.

**Comment by rkiddy** [comment]
I agree. I would just suggest we do not throw away the other data until there is a reason to, If the non-latest adend data can be moved aside so that it does not get in the way of the UI code, that would be great.

**Comment by mikeubell** [comment]
It would be good to not have the obsolete data in the same tables with the data being queried by UI for performance reasons.  If we don’t provide an interface to it, there is really no reason to store it since it exists at the primary source.  If there is some reason that people would want to query this data then we could have it around.

**Comment by bcipolli** [comment]
Oh man, this is challenging. What's the field that indicates which old transaction data were amended? Is this transaction ID specific to the Socrata system, or is it something that is external to Socrata (like `FilerID` or `fips_id`)?
